src/metabase/sync/analyze/classifiers/category.clj

```diff
@@ -9,22 +9,31 @@
             [schema.core :as s]))
 
 
-(s/defn ^:private cannot-be-category? :- s/Bool
-  [base-type :- su/FieldType]
-  (or (isa? base-type :type/DateTime)
-      (isa? base-type :type/Collection)))
+(def ^:const ^:Integer category-string-length-threshold
+  "Maximum average string length for a field to be considered a category."
+  20)
+
+(s/defn ^:private category-candidate? :- s/Bool
+  [{:keys [base_type special_type database_type]}]
+  (and (nil? special_type)
+       (or (isa? base_type :type/Integer)
+           (isa? base_type :type/Text))
+       ;; Druid does not support group by on metrics so these should never be
+       ;; marked as category.
```

Comment:
I'd also note that a metric (i.e. a number mean to be interpreted, added, etc) seems to intrinsically not be a category